docs: Add central SQL statement compatibility matrix by connector (#27824)
prestodb/presto: presto-docs/src/main/sphinx/connector.rst

This chapter describes the connectors available in Presto to access data
from different data sources.

> See :doc:`sql/connector-statement-support` for a comprehensive overview and compatibility matrix of SQL statement support across all connectors.

Reviewer: ```suggestion
See :doc:`sql/connector-statement-support` for an overview and compatibility matrix of SQL statement support across all connectors.
```
